fe/fe-core/src/main/java/com/starrocks/external/RemoteScanRangeLocations.java

```diff
@@ -90,12 +88,18 @@ private void createScanRangeLocationsForSplit(long partitionId,
         hdfsScanRange.setLength(length);
         hdfsScanRange.setPartition_id(partitionId);
         hdfsScanRange.setFile_length(fileDesc.getLength());
-        hdfsScanRange.setFile_format(fileFormat.toThrift());
+        hdfsScanRange.setFile_format(partition.getFormat().toThrift());
         hdfsScanRange.setText_file_desc(fileDesc.getTextFileFormatDesc().toThrift());
         TScanRange scanRange = new TScanRange();
         scanRange.setHdfs_scan_range(hdfsScanRange);
         scanRangeLocations.setScan_range(scanRange);
 
+        if (blockDesc.getReplicaHostIds().length == 0) {
+            String message = String.format("hdfs file block has no host. file = %s/%s",
+                    partition.getFullPath(), fileDesc.getFileName());
+            throw new RuntimeException(message);
```

Comment:
why throw RuntimeException?

Reply:
Because there is no machine to host that block, and there is no way to scan that block. What's your suggestion?

Reply:
RuntimeException may be ignored, I suggest throw another Exception which should be catch explicitly.

Reply:
Nono, that's what I expect. I expect that this exception can throw out and display error mesage like

```
ERROR 1064 (HY000): hdfs file block has no host. file = hdfs://xxxx/stability_data/ssb/orc/part/part-00000-cffe36b9-0225-4ccf-8ea0-71b9c4586233-c000.snappy.orc
```

Otherwise we get error "Unknown error".